src/adapter/src/optimize/peek.rs

```diff
@@ -128,9 +130,10 @@ impl<T> LocalMirPlan<T> {
 
 /// Marker type for [`LocalMirPlan`] structs representing an optimization result
 /// with attached environment context required for the next optimization stage.
-pub struct ResolvedLocal<'s> {
+pub struct ResolvedLocal<'s, 'p> {
     stats: Box<dyn StatisticsOracle>,
     session: &'s Session,
+    peek_ctx: &'p PeekContext,
 }
```

Comment:
Just driving by, but you probably don't need these to be different lifetimes. Does this compile if you use `'s` for both? If so, that'll make the code simpler. Though would recommend renaming the lifetime to `'a` rather than `'s` because `'s` looks like it's named for *s*ession and if you're using it for both *s*ession and *p*eek context, better to use a generic name like `'a`.

Reply:
ah updated, ty